metrics/src/recorder/mod.rs

```diff
@@ -57,6 +57,95 @@ pub trait Recorder {
     fn register_histogram(&self, key: &Key, metadata: &Metadata<'_>) -> Histogram;
 }
 
+// Blanket implementations.
+
+impl<T> Recorder for Rc<T>
+where
+    T: Recorder,
+{
+    fn describe_counter(&self, key: KeyName, unit: Option<Unit>, description: SharedString) {
+        Deref::deref(self).describe_counter(key, unit, description)
```

Comment:
You should be able to just do `(**self)`, or even `self.as_ref()`, which I think are both a bit more idiomatic than `Deref::deref(self)`. Dealer's choice on which one you choose... but just not `Deref::deref(self)`. 😂